sentry/src/main/java/io/sentry/SentryOptions.java

```diff
@@ -354,6 +354,31 @@ public class SentryOptions {
   /** ClientReportRecorder to track count of lost events / transactions / ... * */
   @NotNull IClientReportRecorder clientReportRecorder = new ClientReportRecorder(this);
 
+  /**
+   * Controls how many seconds to wait for sending events in case there were Startup Crashes in the
+   * previous run. Sentry SDKs normally send events from a background queue, but in the case of
+   * Startup Crashes, it blocks the execution of the {@link Sentry#init()} function for the amount
+   * of startupCrashFlushTimeoutMillis to make sure the events make it to Sentry.
+   *
+   * When the timeout is reached, the execution will continue on background.
+   *
+   * Default is 5000 = 5s.
+   */
+  private long startupCrashFlushTimeoutMillis = 5000; // 5s
+
+  /**
+   * Controls the threshold after the application startup time, within which a crash should happen
+   * to be considered a Startup Crash.
+   *
+   * Startup Crashes are sent on @{@link Sentry#init()} in a blocking way, controlled by
+   * {@link SentryOptions#startupCrashFlushTimeoutMillis}.
+   *
+   * Default is 2000 = 2s.
+   */
+  private long startupCrashDurationThresholdMillis = 2000; // 2s
+
+  private long sdkInitTime = 0;
```

Comment:
Consider refactoring this, `SentryPerformanceProvider` and `SentryAndroid` already set:
`AppStartState.getInstance().setAppStartTime(appStartMillis, appStartTime);`
Which should be pretty much similar, maybe reusing would make sense, if possible.

Reply:
Yeah, it makes sense. At first I thought I will also cover desktop/backend apps, but then I realized that `SendCachedFireAndForget` is only applied for Android.

afaiu, desktop and backend apps don't need this, because they are covered by the `ShutdownHookIntegration`, which will run a blocking flush before terminating, so we're good there as well. 

I will change it to the `AppStartState` things here 👍

Reply:
Makes sense.